MissingObjectException on tree iterator

## Expected Behavior
Queries working on all tree entries for our siva datasets.

## Current Behavior
Query:

```sql
SELECT
    token,
    COUNT(DISTINCT repository_id) AS repository_count
FROM (
    SELECT
        b.repository_id,
        EXPLODE(SPLIT(b.content, '[^_A-Za-z0-9]+')) AS token
    FROM
        blobs b
        JOIN repository_ids_no_forks r ON
            b.repository_id = r.repository_id
    WHERE
        reference_name = 'refs/heads/HEAD' AND
        is_binary = false AND
        LENGTH(content) < 524288
) AS q
GROUP BY token
ORDER BY repository_count DESC
LIMIT 50
```

Dataset: `/100-siva` (in our staging cluster).

Crash:

```
Job aborted due to stage failure: Task 2 in stage 28.0 failed 1 times, most recent failure: Lost task 2.0 in stage 28.0 (TID 2216, localhost, executor driver): tech.sourced.engine.iterator.RepositoryException: Repository error with data: Repository[/tmp/spark-c3e0aed7-dd88-448b-bfd4-22aec9cc156e/processing-repositories/B25E14E8CC093B899869839601B14C54/386bc1ad519609dc6ece05296a80c003d4c8e107.siva]; urls git://github.com/WarlordCore/Justice-Core, git@github.com:WarlordCore/Justice-Core.git, https://github.com/WarlordCore/Justice-Core.git
Caused by: org.eclipse.jgit.errors.MissingObjectException: Missing tree 0ba835cd5bafc7ed2ccf90d67b882a696af51771
	at org.eclipse.jgit.internal.storage.file.WindowCursor.open(WindowCursor.java:164)
	at org.eclipse.jgit.treewalk.CanonicalTreeParser.reset(CanonicalTreeParser.java:214)
	at org.eclipse.jgit.treewalk.TreeWalk.parserFor(TreeWalk.java:1347)
	at org.eclipse.jgit.treewalk.TreeWalk.addTree(TreeWalk.java:741)
	at tech.sourced.engine.iterator.GitTreeEntryIterator$.tech$sourced$engine$iterator$GitTreeEntryIterator$$getTreeEntries(GitTreeEntryIterator.scala:146)
	at tech.sourced.engine.iterator.GitTreeEntryIterator$$anonfun$1.apply(GitTreeEntryIterator.scala:124)
	at tech.sourced.engine.iterator.GitTreeEntryIterator$$anonfun$1.apply(GitTreeEntryIterator.scala:124)
```

## Possible Solution
Similar to the other iterator fixes for jgit exceptions.

## Steps to Reproduce
1. Execute the above query in our cluster.

## Context
Two of our (tentative) reference SQL queries give this error with a relatively small dataset found in our staging cluster.

## Your Environment
* Version used: plain jupyter + `---packages tech.sourced:engine:0.5.7`
* Operating System and version: kubernetes (staging cluster)

--- a/src/main/scala/tech/sourced/engine/iterator/GitTreeEntryIterator.scala
+++ b/src/main/scala/tech/sourced/engine/iterator/GitTreeEntryIterator.scala
@@ -1,10 +1,13 @@
 package tech.sourced.engine.iterator
 
 import org.apache.spark.internal.Logging
-import org.eclipse.jgit.errors.{IncorrectObjectTypeException, MissingObjectException}
+import org.eclipse.jgit.errors.{
+  CorruptObjectException,
+  IncorrectObjectTypeException,
+  MissingObjectException
+}
 import org.eclipse.jgit.lib.{ObjectId, Repository}
 import org.eclipse.jgit.treewalk.TreeWalk
-import tech.sourced.engine.iterator.GitTreeEntryIterator.log
 import tech.sourced.engine.util.{CompiledFilter, Filters}
 
 abstract class TreeEntryIterator(finalColumns: Array[String],
@@ -121,7 +124,17 @@ object GitTreeEntryIterator extends Logging {
       )
     }
 
-    var iter: Iterator[TreeEntry] = commits.flatMap(c => getTreeEntries(repo, c))
+    var iter: Iterator[TreeEntry] = commits.flatMap(c =>
+      try {
+        getTreeEntries(repo, c)
+      } catch {
+        case e@(
+          _: MissingObjectException |
+          _: IncorrectObjectTypeException |
+          _: CorruptObjectException) =>
+          log.debug(s"incorrect tree entry found for ${RepositoryException.repoInfo(repo)}", e)
+          Nil
+      })
     if (filters.hasFilters("path")) {
       iter = iter.filter(te => filters.matches(Seq("path"), te.path))
     }

